pkgs/development/libraries/flatpak/binary-path.patch

```diff
@@ -0,0 +1,29 @@
+diff --git a/common/flatpak-dir.c b/common/flatpak-dir.c
+index eba81fef..134024e2 100644
+--- a/common/flatpak-dir.c
++++ b/common/flatpak-dir.c
+@@ -7532,8 +7532,13 @@ export_desktop_file (const char         *app,
+       g_key_file_remove_key (keyfile, groups[i], "X-GNOME-Bugzilla-ExtraInfoScript", NULL);
+ 
+       new_exec = g_string_new ("");
+-      if ((flatpak = g_getenv ("FLATPAK_BINARY")) == NULL)
+-        flatpak = FLATPAK_BINDIR "/flatpak";
++      if (g_str_has_suffix (name, ".service"))
++      {
++        flatpak = "/run/current-system/sw/bin/flatpak";
++      } else {
++        if ((flatpak = g_getenv ("FLATPAK_BINARY")) == NULL)
++          flatpak = "flatpak";
++      }
```

Comment:
I think this could be made to work on both nixos and non if the existence of `/run/...` path is checked

Reply:
I could modify it to this: https://github.com/vlinkz/nixpkgs/commit/eabde3df7d442d583502abd9a0ffe4d52a780c96, however flatpak still won't work on non-NixOS as `flatpak-system-helper.service` needs to be startable, however the services are stuck in `~/.nix-profile/lib/systemd/{user,system}`.

Reply:
someone could symlink that but hopefully no one will ever do that